### PR TITLE
Enable local variable shadowing warnings

### PR DESCRIPTION
**Pull request by slipher**

### CODE REVIEW

File: src/common/math/Vector.h [hunk 1]
```diff
@@ -876,17 +876,6 @@ namespace Math {
 		return init;
 	}
 
-	// Vector minimum/maximum and clamping
```
Comment (DolceTriade) [inline src/common/math/Vector.h:879]: Unused I assume?

**Review by DolceTriade — COMMENTED** [review]
CI says there was an instance missed

```
/usr/bin/g++ -DBUILD_ENGINE -DDEBUG_BUILD -DNACL_ANDROID=0 -DNACL_BUILD_ARCH=x86 -DNACL_BUILD_SUBARCH=64 -DNACL_LINUX=1 -DNACL_OSX=0 -DNACL_WINDOWS=0 -DUSE_CURSES -I/home/vsts/work/1/s/src -I/home/vsts/work/1/s/libs -I/home/vsts/work/1/s/build/CMakeFiles -I/home/vsts/work/1/s/libs/nacl -I/usr/include/ncursesw -I/usr/include/SDL2 -I/usr/include/opus -I/usr/include/freetype2 -I/usr/include/AL -I/home/vsts/work/1/s/src/engine -ffast-math -fno-strict-aliasing -m64 -mtune=generic -mcx16 -fvisibility=hidden -fno-gnu-unique -std=gnu++14 -pthread -g -g3 -Wall -Wextra -Werror -Wshadow=local -Wno-pragmas -Wno-unknown-pragmas -Wno-error=old-style-cast -Woverloaded-virtual -Wstrict-null-sentinel -Wno-error=sign-promo -Wno-error=stringop-overflow= -Wno-error=maybe-uninitialized -Wno-error=missing-field-initializers -Wno-error=sign-compare -Wno-error=switch -Wno-error=unused-but-set-parameter -Wno-error=unused-but-set-variable -Wno-error=unused-function -Wno-error=unused-parameter -Wno-error=unused-variable -Wno-error=format-truncation -MD -MT CMakeFiles/engine-lib.dir/src/engine/framework/System.cpp.o -MF CMakeFiles/engine-lib.dir/src/engine/framework/System.cpp.o.d -o CMakeFiles/engine-lib.dir/src/engine/framework/System.cpp.o -c /home/vsts/work/1/s/src/engine/framework/System.cpp
/home/vsts/work/1/s/src/engine/framework/System.cpp: In function ‘void Sys::SignalHandler(int)’:
/home/vsts/work/1/s/src/engine/framework/System.cpp:360:11: error: declaration of ‘int sig’ shadows a parameter [-Werror=shadow=compatible-local]
  for (int sig: {SIGTERM, SIGINT, SIGQUIT, SIGPIPE, SIGHUP})
           ^~~
/home/vsts/work/1/s/src/engine/framework/System.cpp:341:31: note: shadowed declaration is here
 static void SignalHandler(int sig)
                               ^~~
```

File: src/common/math/Vector.h (hunk 1 above)
Comment (slipher) [reply src/common/math/Vector.h:879]: There was one solitary use case of Distance which I moved to the vec3_t API.

**Comment by slipher** [comment]
Squashed some Linux-specific ones and a debug build-specific one. I had to make the temporary variables in the assert macros use funny names so that they wouldn't shadow anything.

**Pull request merged**